### PR TITLE
Forecast V2 Fast Explainability

### PR DESCRIPTION
**Pull request by prasankh**
[ODSC-50379](https://jira.oci.oraclecorp.com/browse/ODSC-50379)
* Introduced three modes of explainability HIGH_ACCURACY, BALANCED and FAST_APPROXIMATE.
* Moved away from KernelExplainer to PermutationExplainer to generate explanations faster.

### CODE REVIEW

File: ads/opctl/operator/lowcode/forecast/schema.yaml [hunk 1]
```diff
@@ -234,6 +234,15 @@ spec:
       meta:
         description: "Explainability, both local and global, can be disabled using this flag. Defaults to false."
 
+    explanations_accuracy_mode:
+      type: string
+      required: false
+      default: FAST_APPROXIMATE
```
Comment (ahosler) [inline ads/opctl/operator/lowcode/forecast/schema.yaml:240]: I like this as the default 👍

File: ads/opctl/operator/lowcode/forecast/model/automlx.py [hunk 2]
```diff
@@ -316,7 +316,6 @@ def _custom_predict_automlx(self, data):
         -------
             numpy.ndarray: The predicted future values of the time series.
         """
-        temp = 0
         data_temp = pd.DataFrame(
             data,
             columns=[col for col in self.dataset_cols],
```
Comment (ahosler) [inline ads/opctl/operator/lowcode/forecast/model/automlx.py:321]: We don't need the "unit='s'" here, correct?

File: ads/opctl/operator/lowcode/forecast/model/base_model.py [hunk 3]
```diff
@@ -574,49 +574,37 @@ def explain_model(self, datetime_col_name, explain_predict_fn) -> dict:
...
+            data_trimmed = data.tail(max(int(len(data) * ratio), 100)).reset_index()
+            data_trimmed[datetime_col_name] = data_trimmed[datetime_col_name].apply(lambda x: x.timestamp())
+            kernel_explnr = PermutationExplainer(
                 model=explain_predict_fn,
-                data=self.bg_data[list(self.dataset_cols)][: -self.spec.horizon][
-                    list(self.dataset_cols)
-                ],
+                masker=data_trimmed,
                 keep_index=False
                 if self.spec.model == SupportedModels.AutoMLX
```
Comment (ahosler) [inline ads/opctl/operator/lowcode/forecast/model/base_model.py:603]: Not necessary for this PR, but we should make this a property of the model that gets re-defined in automlx

**Comment by ahosler** [comment]
image
This looks great! Let's merge

**Review by ahosler — APPROVED** [review]

File: ads/opctl/operator/lowcode/forecast/model/automlx.py (hunk 2 above)
Comment (prasankh) [reply ads/opctl/operator/lowcode/forecast/model/automlx.py:321]: Right.

**Pull request merged**